### PR TITLE
Add "kfp run delete <run-id>" to CLI.

### PR DESCRIPTION
**Pull request by jingzhang36**
This is part of api unification 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1646)

### CODE REVIEW

**Comment by Ark-kun** [comment]
AFAIK, I've already exposed all APIs.

`client.runs.delete_run()`

**Comment by jingzhang36** [comment]
> AFAIK, I've already exposed all APIs.
> 
> `client.runs.delete_run()`

Can you post a link for my reference? Thanks! I searched this string "client.runs.delete_run", but couldn't find any occurrence in kpf repo https://github.com/kubeflow/pipelines/search?q="client.runs.delete_run"&unscoped_q="client.runs.delete_run".

**Comment by Ark-kun** [comment]
The support for all APIs was added in https://github.com/kubeflow/pipelines/pull/1536

**Comment by jingzhang36** [comment]
> The support for all APIs was added in #1536

I see. Thanks! Then I can add a CLI command on deleting a run by calling client.runs.delete_run()

**Comment by Ark-kun** [comment]
> > The support for all APIs was added in #1536
> 
> I see. Thanks! Then I can add a CLI command on deleting a run by calling client.runs.delete_run()

What do you think about automatically exposing the commands in CLI the same way? It might be possible to trim suffixes from the methods and have `cli runs delete <run_id>`.

File: sdk/python/setup.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
     install_requires=REQUIRES,
     packages=[
         'kfp',
+        'kfp.cli',
```
Comment (hongye-sun) [inline sdk/python/setup.py]: I have already added it in a PR. You probably need a merge.

File: sdk/python/kfp/cli/run.py [hunk 2]
```diff
@@ -100,4 +100,19 @@ def _display_run(client, namespace, run_id, watch):
...
\ No newline at end of file
+    print(tabulate(data, headers=headers, tablefmt='grid'))
+
+@run.command()
+@click.argument('run-id')
+@click.pass_context
+def delete(ctx, run_id):
+    """delete a KFP run"""
+    client = ctx.obj['client']
+    # KFP backend API DeleteRun() returns google.protobuf.Empty
```
Comment (hongye-sun) [inline sdk/python/kfp/cli/run.py]: Could you add a delete_run method in the client with the try-catch? I want to minimize the client logic in cli.

File: sdk/python/setup.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/setup.py]: GitHub does not consider this a conflict though...

File: sdk/python/kfp/cli/run.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/kfp/cli/run.py]: This is a bug in swagger codegen and we should try to fix it.

Comment (Ark-kun) [reply sdk/python/kfp/cli/run.py]: I've researched the issue and fixed the problem with swagger codegen. See https://github.com/kubeflow/pipelines/issues/1685

Run `pip3 install kfp-server-api==0.1.18.3` to update the package.

There is no need to use try/catch now.

Comment (hongye-sun) [reply sdk/python/kfp/cli/run.py]: It's still better to create a delete_run method in the client module for better consistency and discoverability, even though it's just call the runs.delete_run method. I think tfx team also had some issue by directly calling runs.delete_run which causes linter unhappy.

**Pull request closed without merging**